src/commands/watch.rs

```diff
@@ -0,0 +1,71 @@
+
+use db::sea_orm::{EntityTrait, ModelTrait, ActiveValue::Set, ActiveModelTrait};
+use serenity::{
+    framework::standard::{macros::command, Args, CommandResult},
+    model::channel::Message,
+    prelude::Context
+};
+use super::enforce_compliancy;
+use crate::{MemberCache, Connection, commands::{parse_channel, is_moderator}};
+
+#[command] 
+#[min_args(1)]
+#[max_args(2)]
+#[aliases(monitor)]
+async fn watch(ctx: &Context, msg: &Message, mut args: Args) -> CommandResult{
+    if !is_moderator(ctx, msg).await{
+        msg.reply(ctx, "You must be a moderator to run this command").await?;
+        return Ok(())
+    }
+
+
+    let guild_id = *msg.guild_id.unwrap().as_u64() as i64;
+
+    if !enforce_compliancy(ctx, msg).await.0{
```

Comment:
what does the .0 do here - just noticed it's different from what the is_moderator check is doing

Reply:
enforce_compliancy in `src/commands/mod.rs` returns a future of `(bool, Option<db::guild::Model>)`
bool is whether the guild is compliant.
`(a, b, c).0 = a`

Reply:
ah i see cool

Reply:
now i think about it, this shouldnt be necessary as it can just return an Option<db::guild::Model> and you can trivially get the compliancy from that.

Reply:
oh true.  good catch